src/parser/ad_parser.js

```diff
@@ -232,47 +232,62 @@ function parseWrapper(wrapperElement) {
  */
 function parseExtensions(collection, extensions) {
   extensions.forEach(extNode => {
-    const ext = new AdExtension();
-    const extNodeAttrs = extNode.attributes;
-    const childNodes = extNode.childNodes;
+    const ext = parseExtension(extNode);
 
-    if (extNode.attributes) {
-      for (let extNodeAttrKey in extNodeAttrs) {
-        const extNodeAttr = extNodeAttrs[extNodeAttrKey];
-
-        if (extNodeAttr.nodeName && extNodeAttr.nodeValue) {
-          ext.attributes[extNodeAttr.nodeName] = extNodeAttr.nodeValue;
-        }
-      }
+    if (ext) {
+      collection.push(ext);
     }
+  });
+}
 
-    for (let childNodeKey in childNodes) {
-      const childNode = childNodes[childNodeKey];
-      const txt = parserUtils.parseNodeText(childNode);
-
-      // ignore comments / empty value
-      if (childNode.nodeName !== '#comment' && txt !== '') {
-        const extChild = new AdExtensionChild();
-        extChild.name = childNode.nodeName;
-        extChild.value = txt;
-
-        if (childNode.attributes) {
-          const childNodeAttributes = childNode.attributes;
-
-          for (let extChildNodeAttrKey in childNodeAttributes) {
-            const extChildNodeAttr = childNodeAttributes[extChildNodeAttrKey];
+/**
+ * Parses an extension child node
+ * @param {Object} extNode - The extension node to parse
+ * @param {Boolean} isChild - A flag to tell how to parse the node
+ * @return {Object}
+ */
+function parseExtension(extNode, isChild = false) {
```

Comment:
There seems to be a problem with each iteration of a node in the extension. For each iteration, it adds an extra:
```
{
    "name": null,
    "value": null,
    "attributes": {},
    "children": [
         {
              "value": 0,
              "attributes": {},
               "children": []
         }
     ]
},
```